test/index.ts

```diff
@@ -1,25 +1,51 @@
-import { fastify } from 'fastify';
-import { runServer } from '../src/server';
-import dotenv from 'dotenv';
 import axios from 'axios';
+import * as chai from 'chai';
+import prismaInstance from '../src/db';
+import { closeServer, runServer } from '../src/server';
+import { UserReq } from '../src/user/user.model';
 
-dotenv.config({ path: '../.test.env' });
-const app = fastify();
-const port: number = process.env.PORT || 8081;
-const server = axios.create({
+const port: number = process.env.PORT ? parseInt(process.env.PORT, 10) : 8081;
+const prisma = prismaInstance;
+const endpoint = axios.create({
   baseURL: `http://localhost:${port}`,
 });
+const input: UserReq = {
+  name: 'Gabrielle',
+  email: 'gabi@gmail.com',
+  password: 'coxinha123',
+  birthDate: '02-12-2000',
+};
 
-before(() => {
-  app.get('/hello', () => {
-    return 'hello, world!';
-  });
-  runServer(app, port);
+before(async () => {
+  try {
+    await prisma.$connect();
+    await runServer(port);
+  } catch (error) {
+    console.error(error);
+    process.exit(1);
+  }
 });
 
 describe('Hello, world endpoint', () => {
-  it('return hello, world', async () => {
-    const response = await server.get('/hello');
-    console.log(response.data);
+  it(`return "hello, world!"`, async () => {
+    const response = await endpoint.get('/hello');
+    chai.expect(response.data).to.be.eq('hello, world!');
+  });
+});
+
+describe('Create user endpoint', () => {
+  it('Response sucess', async () => {
+    const response = await endpoint.post('/users', input);
+    chai.expect(response.status).to.be.eq(201);
+    chai.expect(response.data).to.have.deep.property('id');
+    chai.expect(response.data.name).to.be.deep.eq(input.name);
+    chai.expect(response.data.email).to.be.deep.eq(input.email);
+    chai.expect(response.data.birthDate).to.be.deep.eq(input.birthDate);
+    chai.expect(response.data).to.not.have.property('password');
```

Comment:
Tem uma forma mais completa de testar bodiues de resposta em REST, que é usar o `to.be.deep.eq(...)`. Ele checa dois objetos de forma bem minunciosa, todas as chaves e valores, e inclusive dá erro se tiver faltando ou sobrando campos. É importante para que a gente tenha certeza de que está retornando apenas do que queremos que retorne mesmo.